### DISCUSSION TITLE
Uncaught Error: No QueryClient set, use QueryClientProvider to set one when using hook useEnsName, useSignMessage and useConnect

### DISCUSSION THREAD

**Discussion by nadiia-balaian** [wevm/wagmi#3481] [Q&A]
I used react-query in my application, but version 3.x.x.

I decided to migrate from wagmi 1.x to wagmi 2.x. After that I have to make migration to react-query 5.x. So I expect everything should work fine, cause I already had QueryClientProvider.

Now application is wrapped with WagmiProvider and QueryClientProvider.
When I use hook useAccount application works fine, but when I add useConnect, useSignMessage or useEnsName application crashes with an error.

I tried to use different order, tried to wrap components separately.
I would appreciate help or hint what else should I check.

<img width="441" alt="Screenshot 2024-01-18 at 14 48 35" src="https://github.com/wevm/wagmi/assets/109521832/560f83c7-7768-4a6e-9c53-27716f05a7b4">

<img width="500" alt="Screenshot 2024-01-18 at 14 49 05" src="https://github.com/wevm/wagmi/assets/109521832/a35a9a58-b314-4a63-bfc7-981ffa34abb6">

**Comment by Yuripetusko** [comment]
I had the same error today and turns out I had `@tanstack/react-query-devtools` installed that I forgot to update to version 5, and additionally as I was moving QueryCLientProvider inside of WagmiProvider, my `ReactQueryDevtools` was left outside of react query provider (it should be inside for context to pass). 

Perhaps you have a similar issue?

**Reply by nadiia-balaian** [reply]
Thanks for your answer, but I don't use @tanstack/react-query-devtools in my project. so probably issue is related to something else. I decided to rollback to previous version :(

**Reply by tmm** [reply]
If you provide a minimal reproduction, I'm happy to take a look. The screenshot alone is not useful enough to look into what is going on.

**Comment by haroon-schmrkow** [comment]
We are facing the same issue, crash due to lack of provider (which should be there) after updating to wagmi `2.5.18` and react-query `5.28.14`

**Reply by tmm** [reply]
Once again, please provide a minimal reproduction if you want anyone to help you out. The info you provided is not useful.
